microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/faulttolerance/java/MicroProfileFaultToleranceASTValidator.java

```diff
@@ -115,14 +123,24 @@ public boolean visit(TypeDeclaration type) {
 			if (modifier instanceof Annotation) {
 				Annotation annotation = (Annotation) modifier;
 				if (isMatchAnnotation(annotation, ASYNCHRONOUS_ANNOTATION)) {
-					try {
-						MethodDeclaration[] methods = type.getMethods();
-						for (MethodDeclaration node : methods) {
-							validateAsynchronousAnnotation(node, (MarkerAnnotation) modifier);
+					BiConsumer<MethodDeclaration, Annotation> validateAsynchronous = (scanType, scanAnnotation) -> {

```

Comment:
Is there any reason to do it this way ? Why not just call both methods here.

Reply:
The original intention was to reduce the repetitive code for iterating through the methods when the annotation was declared in the class. However, given `@Retry` doesn't interact with methods like `@Asynchronous` does, I don't believe I need it.

Reply:
I see your point. It saves having to write out the for-loop in each condition, but you still end up writing out the method declaration that gets passed it. I just found the code a bit more confusing because you have to follow the calls to ultimately figure out that accept(T, U) of the BiConsumer finally does the validation.

I thought just calling directly would probably be easier to read.